scipy/sparse/_sputils.py

```diff
@@ -199,12 +200,12 @@ def get_index_dtype(arrays=(), maxval=None, check_contents=False):
     return dtype
 
 
-def get_sum_dtype(dtype):
+def get_sum_dtype(dtype: np.dtype) -> np.dtype:
     """Mimic numpy's casting for np.sum"""
-    if dtype.kind == 'u' and np.can_cast(dtype, np.uint):
-        return np.uint
-    if np.can_cast(dtype, np.int_):
-        return np.int_
+    if dtype.kind == 'u' and np.can_cast(dtype, np_ulong):
+        return np.dtype(np_ulong)
+    if np.can_cast(dtype, np_long):
+        return np.dtype(np_long)
```

Comment:
This is actually *not* correct because long just happens to be the default integer dtype!  It should still use `np.dtype(int)` or `np.int_`.

Reply:
We had discussed many times that we do want (or at least may want) to change the default integer!

Reply:
Addressed in https://github.com/scipy/scipy/pull/19394.